x-pack/legacy/plugins/siem/server/lib/detection_engine/alerts/utils.ts

```diff
@@ -152,11 +154,22 @@ export const singleBulkIndex = async ({
   if (someResult.hits.hits.length === 0) {
     return true;
   }
+  // index documents after creating an ID based on the
+  // source documents' originating index, and the original
+  // document _id. This will allow two documents from two
+  // different indexes with the same ID to be
+  // indexed, and prevents us from creating any updates
+  // to the documents once inserted into the signals index,
+  // while preventing duplicates from being added to the
+  // signals index if rules are re-run over the same time
+  // span. Also allow for versioning.
   const bulkBody = someResult.hits.hits.flatMap(doc => [
     {
-      index: {
+      create: {
         _index: signalsIndex,
-        _id: doc._id,
+        _id: createHash('sha256')
+          .update(doc._index.concat(doc._id, doc?._version ?? ''))
+          .digest('hex'),
```

Comment:
I would push this into a small utility function and add tests so it makes it easy to understand the intent based on the different input scenarios for tests.